### DISCUSSION TITLE
Redirect Thoughts

### DISCUSSION THREAD

**Discussion by ronag** [nodejs/undici#2230] [General]
Thoughts regarding our current redirect handler:

- History does not allow a temporary redirect to redirect back to server. Seems like a bit of an edge-case but I think this should be allowed?
- We are missing a redirect cache/store where subsequent requests to a redirected url follow previous redirects?
- Is there any semi-standard header that the client can specify to whether or not it allows for redirects or wants e.g. a reverse proxy to follow the redirect for the client?

**Comment by ronag** [comment]
@KhafraDev

**Comment by metcoder95** [comment]
>History does not allow a temporary redirect to redirect back to server. Seems like a bit of an edge-case but I think this should be allowed?

Do you mean a full roundtrip? it should be possible, isn't it? Or do you refer to something else?
By spec, is needed to detect and intervene on cyclical redirections.

>We are missing a redirect cache/store where subsequent requests to a redirected url follow previous redirects?

This can be helpful. And would be needed to have control over it as possible redirects are just ephemeral and on a subsequent request the redirect is gone, and a single request reaches your destination (maybe an edge case but possibly including in-between hops), unless the status code is `308` or `301`, then is safe to make such assumption.

>Is there any semi-standard header that the client can specify to whether or not it allows for redirects or wants e.g. a reverse proxy to follow the redirect for the client?

Over a quick Google search, it seems is not, but maybe didn't reach the right resource

On a side note, I think the `Redirect Handler` does also not consider possible method changes, which are possible through `301`, `302`, and `307`. But maybe not worth it?
